backend/pkg/database/tracesampling.go

```diff
@@ -0,0 +1,77 @@
+// Copyright © 2021 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package database
+
+import (
+	"gorm.io/gorm"
+)
+
+const (
+	traceSamplingTableName = "trace_sampling"
+)
+
+type TraceSampling struct {
+	gorm.Model
+
+	APIID         uint   `json:"api_id,omitempty" gorm:"column:api_id" faker:"-"`
+	TraceSourceID uint   `json:"trace_source_id,omitempty" gorm:"column:trace_source_id" faker:"-"`
+	Component     string `json:"component,omitempty" gorm:"column:component" faker:"-"`
+}
```

Comment:
Need to add a unique index constraint on (APIID, TraceSourceID, Component)

Reply:
Should we add a TraceSourceID as a foreign key ?

Reply:
No Foreign key here, as I can have TraceSourceID==0 as valid value

Reply:
Just seeing your PR 299... Now TraceSourceID==0 is a valid source. BUT:
- in other place (for ex apiinventory.go) TraceSourceID is not a foreign key,
- also... event_id, api_id have never foreign key. 
What missed, if any, is to delete entries from trace_sampling if a TraceSource is deleted